### DISCUSSION TITLE
Streaming upload

### DISCUSSION THREAD

**Discussion by h1vemaster** [aio-libs/aiohttp#6820] [Q&A] [answered]
Is it possible on an aiohttp server to send an uploaded file to another server without loading it into memory? How can I do that?

```python
async def forward_huge_upload_to_upstream(request):
    reader = await request.multipart()
    
    field = await reader.next()
    filename = field.filename

    data = FormData()
    data.add_field('data', ???, filename=field.filename, content_type=field.headers['Conent-Type'])

    async with aiohttp.ClientSession() as session:
        async with session.post(url, data=data) as response:
            return web.Response(status=response.status, body=await response.text())
```

I found [this](https://groups.google.com/g/aio-libs/c/tb3kXb3733s/m/sZjec3_4AwAJ), read a lot of documentation, but didn't understand how to use it.

**Comment by webknjaz** [comment, marked as answer]
The examples in this documentation section https://docs.aiohttp.org/en/stable/client_quickstart.html#streaming-uploads show how to pass a _separate_ async generator function or a pre-opened file handle as a `data`. It also mentions that you should be able to just pass the incoming `request.content` there. https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.content seems to implement the async iterator protocol for this to work.

**Reply by h1vemaster** [reply]
Could you provide a simple example please? I can't understand how to implement async iterator for this.
